### ISSUE TITLE
Sciter is using Kiwi 

### ISSUE THREAD

**Issue by c-smile** [nucleic/kiwi#34]
Just let you know that [Sciter Engine](https://sciter.com/)  is using Kiwi for some flow calculations.

This style:
```
section {
 flow: grid(1 2,
            3 3);
```
with this markup 
```
<section>
   <div>1</div>
   <div>2</div>
   <div>3 (second row, spans two columns)</div>
</section>
```
invokes kiwi constraint solver.

**Comment by sccolbert** [comment]
Thanks! That's pretty awesome.

Do you guys have a custom style parser which hooks all that up? How are you handling resize notifications to know when to invoke the solver?

**Comment by c-smile** [comment]
"custom style parser" 

Sciter uses independent HTML/CSS implementation. So it is 100% "custom".

Solver is invoked each time when width and/or height of such element needs to be recalculated before rendering it.

**Comment by sccolbert** [comment]
> Sciter uses independent HTML/CSS implementation

Holy crap! That's a mammoth amount of effort.

> Solver is invoked each time when width and/or height of such element needs to be recalculated before rendering it.

I so wish we had that in normal browsers :(

**Comment by c-smile** [comment]
"That's a mammoth amount of effort." 

Yes, but last 10 years development was mainly incremental, since first production version appeared as [UI of Norton Antivirus](https://sciter.com/from-skeuomorph-to-flat-ui-evolution-of-one-application/). Architecture did not change that much all these years. Kiwi was added last year as I wanted flexible grid feature to be there.

And here is ["10 years road to Sciter"](https://sciter.com/10-years-road-to-sciter/) article where I explained how I got to that point. Sciter is still mainly one man project.
